GoMeet/src/main/java/com/google/sps/servlets/PopularLocationServlet.java

```diff
@@ -0,0 +1,60 @@
+package main.java.com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.gson.Gson;
+import com.google.sps.data.ServletUtil;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Comparator;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import main.java.com.google.sps.data.Dao;
+import main.java.com.google.sps.data.Location;
+import main.java.com.google.sps.data.LocationDao;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+/** Handles fetching and saving location data. */
+@WebServlet("/popular-location-data")
+public class PopularLocationServlet extends HttpServlet {
+  private Dao<Location> locationDao = new LocationDao();
+
+  /** Responds with a JSON array containing the most popular location data. */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+
+    List<Location> popularLocations = getPopularLocation();
+    String json = ServletUtil.convertToJson(popularLocations);
+    response.getWriter().println(json);
+  }
+
+  /** Returns a list of the locations with the highest vote count. */
+  private List<Location> getPopularLocation() {
+    List<Location> popularLocations = new ArrayList<>();
+    List<Location> allLocations = locationDao.getAll();
+
+    int maxVoteCount = 0;
```

Comment:
I decided to iterate through the list of locations to find the most popular locations as it was:
(1) O(n) each location is visited once.
(2) Given time-constraints, this approach is simple to implement,
(3) In our design doc, we specified that each meeting will have max. 5 locations (so a small set of data).

I considered sorting the list first based on the vote count and then returning the first elements in the list but Collection.sort() has a complexity of O(n*log(n)).

In future, if this feature were to be scaled up, I think storing a list of popular locations as a property of the meeting entity would be more efficient (and updating it when a new location is added) as each location entity for that meeting does not need to be read when requesting the popular locations.